nexus/tests/integration_tests/instances.rs

```diff
@@ -2713,6 +2713,7 @@ async fn test_instances_memory_not_divisible_by_min_memory_size(
     );
 }
 
+#[ignore] // API no longer in sled-agent(-sim). TODO: another way to test w/o propolis?
```

Comment:
We do some Shenanigans to test similar things with crucible from the integration test suite.

If you're running this test suite in a simulated environment, obviously crucible downstairs doesn't exist, because you don't have disks.

Instead, the test suite [directly calls to the sled agent](https://github.com/oxidecomputer/omicron/blob/27dfb62275b92b960a3e2a1c0644cfca492f3faf/nexus/test-utils/src/resource_helpers.rs#L466-L494), calling methods like [`create_zpool` and `create_crucible_dataset`](https://github.com/oxidecomputer/omicron/blob/27dfb62275b92b960a3e2a1c0644cfca492f3faf/sled-agent/src/sim/sled_agent.rs#L269-L281), which just lands in a [simulated implementation of storage](https://github.com/oxidecomputer/omicron/blob/27dfb62275b92b960a3e2a1c0644cfca492f3faf/sled-agent/src/sim/storage.rs). This "simulated implementation" basically just stupidly keeps track of all resources provisioned (it's basically just a few hashmaps of "UUID" -> "what was the request that created this object"). Obviously you can't use this for any real storage, but this means that we can test all the "disk provisioning" stuff in Nexus, and verify what requests *should* have made it down to the sled agent.

Here's the part that I think might be relevant for you:
- In the *real* implementation of Crucible, we expect that each downstairs will be sitting on an HTTP server, responding to requests from Nexus directly....
- ... So in our *simulated* implementation, we [create a server](https://github.com/oxidecomputer/omicron/blob/27dfb62275b92b960a3e2a1c0644cfca492f3faf/sled-agent/src/sim/storage.rs#L384) which [re-implements the crucible endpoints](https://github.com/oxidecomputer/omicron/blob/27dfb62275b92b960a3e2a1c0644cfca492f3faf/sled-agent/src/sim/http_entrypoints_storage.rs) by calling into the "stub implementation".

I suspect we could do something similar for instances:
- When the simulated sled agent should be creating an instance, we could spin up a "simulated" implementation of the propolis server.
- We could implement whatever endpoints would help you write this test, returning whatever data is convenient
- We could expose APIs out of the simulated sled agent to "get direct access to the stub propolis server", so you could check that structure during test.

Reply:
I see! That makes a lot of sense, thanks :D